addons/mail/static/src/messaging/component/activity_mark_done_popover/activity_mark_done_popover_tests.js

```diff
@@ -0,0 +1,285 @@
+odoo.define('mail.messaging.component.ActivityMarkDonePopoverTests', function (require) {
+'use strict';
+
+const components = {
+    ActivityMarkDonePopover: require('mail.messaging.component.ActivityMarkDonePopover'),
+};
+
+const {
+    afterEach: utilsAfterEach,
+    beforeEach: utilsBeforeEach,
+    start: utilsStart,
+} = require('mail.messaging.testUtils');
+
+QUnit.module('mail', {}, function () {
+QUnit.module('messaging', {}, function () {
+QUnit.module('component', {}, function () {
+QUnit.module('ActivityMarkDonePopover', {
+    beforeEach() {
+        utilsBeforeEach(this);
+        this.createActivityMarkDonePopoverComponent = async function (activity) {
+            const ActivityMarkDonePopoverComponent = components.ActivityMarkDonePopover;
+            ActivityMarkDonePopoverComponent.env = this.env;
+            this.component = new ActivityMarkDonePopoverComponent(null, {
+                activityLocalId: activity.localId,
+            });
+            await this.component.mount(this.widget.el);
+        };
+        this.start = async params => {
+            let { env, widget } = await utilsStart(Object.assign({}, params, {
+                data: this.data,
+            }));
+            this.env = env;
+            this.widget = widget;
+        };
+    },
+    afterEach() {
+        utilsAfterEach(this);
+        if (this.component) {
+            this.component.destroy();
+        }
+        if (this.widget) {
+            this.widget.destroy();
+        }
+        this.env = undefined;
+    },
+});
+
+QUnit.test('activity mark done popover simplest layout', async function (assert) {
+    assert.expect(6);
+
+    await this.start();
+    const activity = this.env.entities.Activity.create({
+        activity_category: 'not_upload_file',
+        can_write: true,
+        id:12,
+    });
+    await this.createActivityMarkDonePopoverComponent(activity);
+
+    assert.containsOnce(
+        document.body,
+        '.o_ActivityMarkDonePopover',
+        'Popover component should be present'
```

Comment:
I think we usually double quote the assert last param text (to do everywhere).

Reply:
Right.